apps/language_models/src/pipelines/vicuna_sharded_pipeline.py

```diff
@@ -220,24 +223,43 @@ def compile_to_vmfb(self, inputs, layers, device="cpu", is_first=True):
                 f_.close()
             mlirs.append(bytecode)
 
+        def get_device_index(layer_string):
+            if self.config is None:
+                return None
+            idx_votes = {}
+            for key in self.config.keys():
+                if re.search(layer_string, key):
+                    if int(self.config[key]["gpu"]) in idx_votes.keys():
+                        idx_votes[int(self.config[key]["gpu"])] += 1
+                    else:
+                        idx_votes[int(self.config[key]["gpu"])] = 1
+            device_idx = max(idx_votes, key=idx_votes.get)
+            return device_idx
+
```

Comment:
define function to extract the device index from config file.

The config stores the model in its most granular state, so this function looks for the device used for every layer in the shard, and will use majority vote if different devices are used throughout the shard

Reply:
Elias your comments are really good. But you should put them in the code instead of the review!